### PR TITLE
Continued docs updates

### LINKED ISSUE #10124: master build failing trying to install oshcc and related man pages
See IBM MTT results.  Also I observed this on NERSC system using the nightly tarball.
``` 
Making install in docs
make[1]: Entering directory '/global/u2/h/hpp/mtt_X/master_scratch/MiddlewareGet_OMPIMaster/openmpi-master-202203150241-8c1495d/docs'
make  install-am
make[2]: Entering directory '/global/u2/h/hpp/mtt_X/master_scratch/MiddlewareGet_OMPIMaster/openmpi-master-202203150241-8c1495d/docs'
make[2]: *** No rule to make target 'man-openshmem/man1/oshcc.1', needed by 'all-am'.  Stop.
make[2]: Leaving directory '/global/u2/h/hpp/mtt_X/master_scratch/MiddlewareGet_OMPIMaster/openmpi-master-202203150241-8c1495d/docs'
make[1]: *** [Makefile:2315: install] Error 2
make[1]: Leaving directory '/global/u2/h/hpp/mtt_X/master_scratch/MiddlewareGet_OMPIMaster/openmpi-master-202203150241-8c1495d/docs'
make: *** [Makefile:1469: install-recursive] Error 1
```

### PR DESCRIPTION
**Pull request by jsquyres**
The first commit is some trivial docs updates on the main index page to match the text from the v4.0.x and v4.1.x docs (#10113 and #10112, respectively).

The 2nd commit is worthy of a detailed review: it deals with what version number to prominently display at the top of the front docs index page.  Read the commit message for more details.

**EDIT:** Added a bunch of commits after the above initial description.

Fixes #10124

### CODE REVIEW

**Comment by jsquyres** [comment]
While waiting for a review, I glommed on a few more commits with some more docs updates.  😄

File: docs/installing-open-mpi/compilers-and-flags.rst [hunk 1]
```diff
@@ -22,26 +23,35 @@ line itself The following environment variables are recognized by
 
 .. note:: Open MPI |ompi_ver| does not contain any C++ code.  Hence,
    specifying ``CXXFLAGS`` or ``CXXCPPFLAGS`` is useless (but
-   harmless).  The value of ``CC`` is used as the compiler for the
-   ``mpic++`` wrapper compiler, however.
+   harmless).  The value of ``CXX`` is used as the compiler for the
```
[the remark this answers is not in the document]
Comment (jsquyres) [reply docs/installing-open-mpi/compilers-and-flags.rst]: Discussed verbally -- fixed.

**Comment by ibm-ompi** [comment]
The IBM CI (XL) build failed! Please review the log, linked below.

 Gist: https://gist.github.com/ibm-ompi/88b480fb98dfd2bdac939f023e1f8fe6

**Comment by jsquyres** [comment]
@bwbarrett It took a bunch of tries, but I think I finally got it right: RTD provides some environment variables with the info that we want.  See the "docs: show version at top of first page" commit on this PR.

**Review by bwbarrett — APPROVED** [review]

**Comment by jsquyres** [comment]
bot:aws:retest

**Review by bwbarrett — APPROVED** [review]
I think the makefile changes are ok; I don't see anything wrong, but they kind of hurt my brain (but I think what we're doing is just brain hurting in general, not that there's a problem with this patch).

**Pull request merged**